packages/auth-construct/src/construct.ts

```diff
@@ -1026,30 +1032,23 @@ export class AmplifyAuth
             outputProviders.push('LOGIN_WITH_AMAZON');
           }
           if (providerType === 'OIDC') {
-            outputProviders.push(
-              provider.node['_children']['Resource']['providerName']
-            );
+            outputProviders.push(providerName);
           }
           if (providerType === 'SAML') {
-            outputProviders.push(
-              provider.node['_children']['Resource']['providerName']
-            );
+            outputProviders.push(providerName);
           }
         }
         return JSON.stringify(outputProviders);
       },
     });
 
-    const getUserPoolClientResource = () => {
-      return this.resources.userPoolClient.node['_children']['Resource'];
-    };
-
     output.oauthCognitoDomain = Lazy.string({
       produce: () => {
-        const userPoolDomain =
-          this.resources.userPool.node['_children']['UserPoolDomain'];
+        const userPoolDomain = this.resources.userPool.node.tryFindChild(
+          'UserPoolDomain'
+        ) as UserPoolDomain | undefined;
```

Comment:
We should use typeguards instead and throw errors if the object does not pertain to the type casted.

Reply:
thanks, I updated this to check for types